pkg/bloomtokenizer/bloomtokenizer.go

```diff
@@ -0,0 +1,98 @@
+package bloomtokenizer
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/loki/pkg/chunkenc"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logql/log"
+	"github.com/grafana/loki/pkg/storage/bloom/v1/filter"
+	"github.com/grafana/loki/pkg/storage/chunk"
+	util_log "github.com/grafana/loki/pkg/util/log"
+	"github.com/grafana/loki/tools/tsdb/helpers"
+)
+
+type metrics struct{}
+
+type BloomTokenizer struct {
+	metrics *metrics
+
+	lineTokenizer    Tokenizer
+	chunkIDTokenizer *WrappedTokenizer
+	cache            map[string]interface{}
+}
+
+const CacheSize = 150000
+
+// New returns a new instance of the Bloom Tokenizer.
```

Comment:
Actually if we can add here a two liner of what Tokenizer should do, that'd be very nice. i don't think this is a self-explanatory component.